### PR TITLE
run CI with MariaDB

### LINKED ISSUE #3501: Officially support MariaDB
We could save ourselves a lot of gnashing of teeth during our install and new user onboarding if we could figure out a way to support MariaDB and MySQL simultaneously.  [There is a list of differences between the two RDBs on MariaDBs websites](https://mariadb.com/kb/en/library/incompatibilities-and-feature-differences-between-mariadb-102-and-mysql-57/).

I think the basic approach to getting MariaDB support would be through developing a test suite that runs in multiple environments.  The easiest way to do this would be to make a docker version of BHIMA (#3439), swap out the database, and run the full test suite against it.

### PR DESCRIPTION
**Pull request by individual-it**
fixes #3501

### CODE REVIEW

**Comment by kwilu** [comment]
Hello @individual-it, it seems like you have referenced #3501 in your pull request description, but you have not referenced them in your commit message description(s). When you reference an issue in a commit message, it automatically closes the corresponding issue when the commit is merged.

Please run `git commit --amend` in your command line client to amend your commit message description with `Fixes #3501.`.

An example of a correctly-formatted commit:
```
commit fabd5e450374c8dde65ec35f02140383940fe146
Author: kwilu
Date:   Sat Mar 18 13:42:40 2017 -0700

    pull requests: Check PR commits reference when issue is referenced.

    Fixes #51.
```

Thank you for your contributions to BHIMA!

**Comment by individual-it** [comment]
@kwilu done

**Comment by jniles** [comment]
@individual-it I've been taking a look at the logs, and I'm not sure that the database server is actually changing.  It looks like we'll need to add [mariadb as a Travis Addon](https://docs.travis-ci.com/user/database-setup/#mariadb).  Then we may be able to switch the database server with an environmental variable flag.

I would also throw in [fast_finish](https://docs.travis-ci.com/user/build-matrix/#fast-finishing) since our tests take tens of minutes, this may help clear out the test queue for other tests to run that are enqueued.

**Comment by individual-it** [comment]
You are right, I haven't read the travis docu completely
```
Travis CI does not have any special support for these variables, it just creates three builds with different exported values. It is up to your build script and before_install or before_script steps to make use of them.
```

**Comment by individual-it** [comment]
tests now run with MariaDB, but always fail
image

looks like same issue mentioned in #4335

File: sh/travis.sh [hunk 1]
```diff
@@ -7,6 +7,8 @@ set -a
 source .env.development
 set +a
 
+mysql --version
```
Comment (jniles) [inline sh/travis.sh:10]: Excellent!

File: .travis.yml [hunk 2]
```diff
@@ -12,13 +12,19 @@ env:
     - CHROME_REVISION=706915
     - CHROMEDRIVER_VERSION="79.0.3945.36"
 
+matrix:
+  include:
+    - addons:
+        mariadb: '10.5'
```
Comment (jniles) [inline .travis.yml:18]: :+1:

**Review by jniles — COMMENTED** [review]
@individual-it thanks for your work on this.  It's super exciting to see progress in testing and I'll try to take a look at the flakeyness today.  Hopefully we can come up with either a solution or a work around to address the randomness in it.

**Comment by kwilu** [comment]
Heads up @individual-it, we just merged some commits that conflict with the changes your made in this pull request! You can review this repository's [recent commits](https://github.com/IMA-WorldHealth/bhima/commits/master) to see where the conflicts occur. Please rebase your feature branch against the `upstream/master` branch and [resolve](https://zulip.readthedocs.io/en/latest/git/troubleshooting.html#recover-from-a-git-rebase-failure) your pull request's merge conflicts accordingly.

**Pull request closed without merging**